pkg/operator/encryption/key_controller.go

```diff
@@ -266,22 +266,27 @@ func (c *keyController) getCurrentModeAndExternalReason() (mode, string, error)
 }
 
 // TODO unit tests
-func needsNewKey(grKeys keysState, currentMode mode, externalReason string) (uint64, string, bool) {
-	// if the length of read secrets is more than one (i.e. we have more than just the write key),
-	// then we haven't successfully migrated and removed old keys so you should wait before generating more keys.
-	if len(grKeys.readSecrets) > 1 {
-		return 0, "", false
-	}
-
+func needsNewKey(grKeys keysState, currentMode mode, externalReason string, encryptedGRs []schema.GroupResource) (uint64, string, bool) {
 	// we always need to have some encryption keys unless we are turned off
 	if len(grKeys.readSecrets) == 0 {
 		return 0, "no-secrets", currentMode != identity
 	}
 
 	latestKey, latestKeyID := grKeys.latestKey()
 
+	// if latest secret has been deleted, we will never be able to migrate to that key.
+	if !secretIsBacked(latestKey) {
+		return latestKeyID, "missing-secret", true
```

Comment:
if the key is missing what will be `latestKeyID`?

Reply:
the right one. We have the name, just not the namespace.